examples/wifi/expanded/requests_wifi_api_queuetimes.py

```diff
@@ -0,0 +1,106 @@
+# SPDX-FileCopyrightText: 2024 DJDevon3
+# SPDX-License-Identifier: MIT
+# Coded for Circuit Python 9.x
+"""Queue-Times.com API w/Display Example"""
+
+import os
+import time
+
+import adafruit_connection_manager
+import wifi
+
+import adafruit_requests
+
+# Initalize Wifi, Socket Pool, Request Session
+pool = adafruit_connection_manager.get_radio_socketpool(wifi.radio)
+ssl_context = adafruit_connection_manager.get_radio_ssl_context(wifi.radio)
+requests = adafruit_requests.Session(pool, ssl_context)
+
+# Time between API refreshes
+# 900 = 15 mins, 1800 = 30 mins, 3600 = 1 hour
+SLEEP_TIME = 300
+
+# Get WiFi details, ensure these are setup in settings.toml
+ssid = os.getenv("CIRCUITPY_WIFI_SSID")
+password = os.getenv("CIRCUITPY_WIFI_PASSWORD")
+
+# Publicly Open API (no credentials required)
+QTIMES_SOURCE = "https://queue-times.com/parks/16/queue_times.json"
+
+
+def time_calc(input_time):
+    """Converts seconds to minutes/hours/days"""
+    if input_time < 60:
+        return f"{input_time:.0f} seconds"
+    if input_time < 3600:
+        return f"{input_time / 60:.0f} minutes"
+    if input_time < 86400:
+        return f"{input_time / 60 / 60:.0f} hours"
+    return f"{input_time / 60 / 60 / 24:.1f} days"
+
+
+qtimes_json = {}
+while True:
+    now = time.monotonic()
+    # Connect to Wi-Fi
+    print("\n===============================")
+    print("Connecting to WiFi...")
+    while not wifi.radio.ipv4_address:
+        try:
+            wifi.radio.connect(ssid, password)
+        except ConnectionError as e:
+            print("❌ Connection Error:", e)
+            print("Retrying in 10 seconds")
+    print("✅ WiFi!")
+
+    try:
+        print(" | Attempting to GET Queue-Times JSON!")
+        try:
+            qtimes_response = requests.get(url=QTIMES_SOURCE)
+            qtimes_json = qtimes_response.json()
+        except ConnectionError as e:
+            print("Connection Error:", e)
+            print("Retrying in 10 seconds")
+        print(" | ✅ Queue-Times JSON!")
+
+        DEBUG_QTIMES = False
+        if DEBUG_QTIMES:
+            print("Full API GET URL: ", QTIMES_SOURCE)
+            print(qtimes_json)
+        qtimes_response.close()
+        print("✂️ Disconnected from Queue-Times API")
+
+        print("\nFinished!")
+        print(f"Board Uptime: {time_calc(time.monotonic())}")
+        print(f"Next Update: {time_calc(SLEEP_TIME)}")
+        print("===============================")
+    except (ValueError, RuntimeError) as e:
+        print("Failed to get data, retrying\n", e)
+        time.sleep(60)
+        break
+
+    # Loop infinitely until its time to re-poll
```

Comment:
@FoamyGuy Yes, that is specifically the way they requested it to behave.  

I suppose printing a "last updated" timestamp would make sense but would take up precious screen real estate.  It makes less sense on a huge 3.5" TFT.  

The project intention is for a portable SSD1306 with a lipo battery on a watch band.  In the real script it will attempt to reconnect (via a tethered phone hotspot) every 1 minute.  Perhaps I could improve upon the code comments to specify that's the intention of this particular example?

If you gave one device to each kid and it only worked within range of the hotspot it could ensure that your kids will want to stay close enough to ensure the device works so they get ride updates too.  It's a brilliant incentive for your kids to want to stay close to you in a theme park.

Maybe change the text color after an update so that every new update is unmistakably new?  That would serve as an indicator of new content without taking up any screen real estate.

Reply:
I am still in favor of removing the looping behavior from the example here. That sounds like a neat project and they can definitely still use it that way if they'd like. I could also see a world where someone might want to "favorite" certain rides to get information about them more prominently or even some alerts based on the data returned. 

I believe that in it's usage as an example here the continued printing of the same data over and over makes it more confusing because it's difficult to tell what is happening and when there is actually new data. None of the rest of the examples in this repo behave that way, they're all either "single shot" fetching once and ending, or repeating in a loop that fetches every X minutes or seconds and prints the current data then waits until it's time to run again. I believe it's best to have this example match that behavior and only print the data once after it's been fetched.

Reply:
@FoamyGuy sounds good.  I have the example that the person wanted in my personal repo, and linked them there, and they already have that one.  Turning it into a single use "code done running" sounds good to me.